### PR TITLE
fix(DTFS2-5888-5868): Amendments bug - effective date and first task email

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issues: 
* effective date was not showing on amendments detail page (DTFS2-5868)
* email was not being sent for first task

# Changes made:
* added effective date based on if automatic or manual amendment so is displayed
* updated e2e tests
* added helper for sending first task email
* added api-test

### CODE REVIEW

File: trade-finance-manager-api/src/v1/helpers/amendment.helpers.js [hunk 1]
```diff
@@ -202,9 +205,44 @@ const sendManualBankDecisionEmail = async (amendmentVariables) => {
...
+};
+
+// sends email for first amendment task when pim submit amendment
+const sendFirstTaskEmail = async (taskVariables) => {
+  const { amendment, dealSnapshot, facilityId, amendmentId } = taskVariables;
+  const { tasks } = amendment;
+  const { dealId, exporter, ukefDealId } = dealSnapshot;
+
+  const firstTask = getFirstTask(tasks);
+  const urlOrigin = process.env.TFM_URI;
```
Comment (IgnatG) [inline trade-finance-manager-api/src/v1/helpers/amendment.helpers.js:223]: I'm not sure if this is coming from env variables for the other tasks?

File: trade-finance-manager-api/src/v1/controllers/amendment.controller.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const api = require('../api');
 const { createAmendmentTasks, updateAmendmentTasks } = require('../helpers/create-tasks-amendment.helper');
 const { isRiskAnalysisCompleted } = require('../helpers/tasks');
-const { amendmentEmailEligible, sendAutomaticAmendmentEmail, sendManualDecisionAmendmentEmail, sendManualBankDecisionEmail } = require('../helpers/amendment.helpers');
+const { amendmentEmailEligible, sendAutomaticAmendmentEmail, sendManualDecisionAmendmentEmail, sendManualBankDecisionEmail, sendFirstTaskEmail } = require('../helpers/amendment.helpers');
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/controllers/amendment.controller.js:4]: I normally list functions if more than three in new lines, purely a personal preference not necessary.
Thought I'll add this as a suggestion.

File: e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/add-bank-decision-underwriter-declined.spec.js [hunk 3]
```diff
@@ -168,5 +168,6 @@ context('Amendments underwriting - add banks decision - declined by underwriter'
     facilityPage.facilityTabAmendments().click();
 
     amendmentsPage.amendmentDetails.row(1).bankDecisionTag().contains('Not applicable');
+    amendmentsPage.amendmentDetails.row(1).effectiveDate().should('contain', '-');
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/add-bank-decision-underwriter-declined.spec.js:171]: done

File: trade-finance-manager-api/src/v1/controllers/amendment.controller.js [hunk 4]
```diff
@@ -38,6 +38,11 @@ const sendAmendmentEmail = async (amendmentId, facilityId) => {
           const bankDecisionAmendmentVariables = { user, dealSnapshot, amendment, facilityId, amendmentId };
           await sendManualBankDecisionEmail(bankDecisionAmendmentVariables);
         }
+        // if first amendment task email has not already been sent
+        if (amendment?.sendFirstTaskEmail && !amendment?.firstTaskEmailSent) {
+          const firstTaskVariables = { amendment, dealSnapshot, facilityId, amendmentId };
+          await sendFirstTaskEmail(firstTaskVariables);
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply trade-finance-manager-api/src/v1/controllers/amendment.controller.js:44]: the whole thing is in a try catch block and I have another one in sendFirstTaskEmail  just incase

File: e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/8. amendment-details-manual-approval.spec.js [hunk 5]
```diff
@@ -116,6 +116,7 @@ context('Amendments - Manual approval journey', () => {
 
       facilityPage.facilityTabAmendments().click();
       amendmentsPage.amendmentDetails.row(1).heading().should('contain', 'Amendment 1');
+      amendmentsPage.amendmentDetails.row(1).effectiveDate().should('contain', '-');
```
Comment (abhi-markan) [inline e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/8. amendment-details-manual-approval.spec.js:119]: Missing constant

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Sorry missed out constant on few other places.

File: e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/add-bank-decision-withdraw.spec.js [hunk 6]
```diff
@@ -269,6 +269,7 @@ context('Amendments underwriting - add banks decision - withdraw', () => {
     facilityPage.facilityTabAmendments().click();
 
     amendmentsPage.amendmentDetails.row(1).bankDecisionTag().contains('Withdrawn');
+    amendmentsPage.amendmentDetails.row(1).effectiveDate().should('contain', '-');
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/add-bank-decision-withdraw.spec.js:272]: done

File: e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/8. amendment-details-manual-approval.spec.js (hunk 5 above)
Comment (Zainzzkk) [reply e2e-tests/trade-finance-manager/cypress/integration/journeys/case/amendments/8. amendment-details-manual-approval.spec.js:119]: done

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**